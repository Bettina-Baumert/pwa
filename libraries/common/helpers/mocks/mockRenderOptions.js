import PropTypes from 'prop-types';

const options = {
  context: {
    i18n: () => ({
      __(input) {
        if (input) {
          return input;
        }
        return '';
      },
      _p: () => 'p',
      _d: () => 'd',
      _n: () => 'n',
    }),
    history: {
      length: 0,
      stack: [],
      getActive: () => ({
        key: 'routekey',
      }),
    },
  },
  childContextTypes: {
    i18n: PropTypes.func,
    history: PropTypes.shape(),
  },
};

export default options;
