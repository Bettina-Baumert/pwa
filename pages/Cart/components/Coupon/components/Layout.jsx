import React, { PropTypes } from 'react';
import { Grid } from 'Library/components';
import CouponIcon from './Icon';
import CouponPrice from './Price';
import CouponTitle from './Title';
import CouponCode from './Code';
import CouponDelete from './Delete';
import styles from '../style';

/**
 * The CouponLayout component.
 * @param {Object} props The component properties.
 * @returns {JSX}
 */
const CouponLayout = props => (
  <Grid className={styles.item}>
    <Grid.Item className={styles.icon}>
      <CouponIcon />
    </Grid.Item>
    <Grid.Item className={styles.content} grow={1}>
      <CouponTitle value={props.coupon.label} />
      <CouponCode value={props.coupon.code} />
    </Grid.Item>
    <Grid.Item className={`${styles.content} ${styles.contentLast}`} grow={1} shrink={0}>
      <CouponDelete handleDelete={props.handleDelete} />
      <CouponPrice
        currency={props.currency}
        savedPrice={props.coupon.savedPrice}
      />
    </Grid.Item>
  </Grid>
);

CouponLayout.propTypes = {
  coupon: PropTypes.shape().isRequired,
  currency: PropTypes.string.isRequired,
  handleDelete: PropTypes.func,

};

CouponLayout.defaultProps = {
  handleDelete: () => {},
};

export default CouponLayout;
