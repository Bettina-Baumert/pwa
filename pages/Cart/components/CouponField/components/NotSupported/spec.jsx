import React from 'react';
import { mount } from 'enzyme';
import NotSupported from './index';

describe('NotSupported', () => {
  it('should render as expected without any props', () => {
    const wrapper = mount(<NotSupported />);
    expect(wrapper).toMatchSnapshot();
  });
});
