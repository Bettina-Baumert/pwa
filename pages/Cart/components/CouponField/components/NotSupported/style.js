import { css } from 'glamor';
import variables from 'Styles/variables';
import colors from 'Styles/colors';

export default css({
  background: colors.background,
  display: 'block',
  fontSize: 12,
  margin: '20px 0 -10px',
  padding: `0 ${variables.gap.big}px`,
  textAlign: 'left',
}).toString();
