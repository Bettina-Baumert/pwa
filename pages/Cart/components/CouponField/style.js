import cxs from 'cxs';
import { colors, variables } from 'Templates/styles';

const easing = '450ms cubic-bezier(0.23, 1, 0.32, 1)';

const wrapper = cxs({
  background: colors.light,
  padding: variables.gap.big,
});

const container = cxs({
  position: 'relative',
  height: 24,
  width: '100%',
  fontSize: '0.875rem',
});

const icon = cxs({
  color: colors.primary,
  fontSize: '1.875rem',
  position: 'relative',
  float: 'right',
  top: -10,
  transition: `opacity ${easing}`,
});

const underlineWrapper = cxs({
  position: 'absolute',
  width: '100%',
  borderBottom: `1px solid ${colors.shade5}`,
  top: 24,
});

const underline = cxs({
  position: 'relative',
  width: '100%',
  top: 1,
  height: 2,
  willChange: 'transform',
  transition: `transform ${easing}`,
  background: colors.primary,
});

const underlineBlurred = cxs({
  transform: 'scale3d(0,1,1)',
});

const label = cxs({
  position: 'absolute',
  pointerEvents: 'none',
  bottom: 2,
  color: colors.shade4,
});

const input = cxs({
  position: 'absolute',
  width: 'calc(100% - 36px)',
  top: 0,
  outline: 0,
  bottom: 6,
});

export default {
  wrapper,
  container,
  icon,
  underlineWrapper,
  underline,
  underlineBlurred,
  label,
  input,
};
