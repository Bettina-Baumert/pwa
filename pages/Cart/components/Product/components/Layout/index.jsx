/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@shopgate/pwa-common/components/Grid';
import ProductImage from 'Components/ProductImage';
import QuantityPicker from 'Components/QuantityPicker';
import Title from '../Title';
import ProductPrice from '../ProductPrice';
import Properties from '../Properties';
import styles from './style';

/**
 * The ProductLayout component.
 * @param {Object} props The component properties.
 * @returns {JSX}
 */
const ProductLayout = props => (
  <Grid className={styles.item}>
    <Grid.Item className={styles.leftColumn}>
      <div className={styles.image}>
        <ProductImage src={props.product.featuredImageUrl} />
      </div>
      <QuantityPicker
        quantity={props.quantity}
        editMode={props.editMode}
        onChange={props.handleUpdate}
        onToggleEditMode={props.toggleEditMode}
      />
    </Grid.Item>
    <Grid.Item className={styles.content} grow={1}>
      <Title
        handleRemove={props.handleDelete}
        toggleEditMode={props.toggleEditMode}
        value={props.product.name}
      />
      <Grid className={styles.info}>
        <Properties
          properties={props.product.properties}
        />
        <ProductPrice
          currency={props.currency}
          defaultPrice={props.product.price.default}
          specialPrice={props.product.price.special}
        />
      </Grid>
    </Grid.Item>
  </Grid>
);

ProductLayout.propTypes = {
  currency: PropTypes.string.isRequired,
  editMode: PropTypes.bool.isRequired,
  product: PropTypes.shape().isRequired,
  quantity: PropTypes.number.isRequired,
  handleDelete: PropTypes.func,
  handleUpdate: PropTypes.func,
  toggleEditMode: PropTypes.func,
};

ProductLayout.defaultProps = {
  handleDelete: () => {},
  handleUpdate: () => {},
  toggleEditMode: () => {},
};

export default ProductLayout;
