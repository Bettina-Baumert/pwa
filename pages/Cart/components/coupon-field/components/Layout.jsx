import React, { PropTypes } from 'react';
import {
  Input,
  I18n,
} from 'Library/components';
import AddIcon from './Icon';
import styles from '../style';

/**
 * The CouponFieldLayout component.
 * @param {Object} props The component properties.
 * @returns {JSX}
 */
const CouponFieldLayout = props => (
  <div className={styles.wrapper}>
    <form
      className={styles.container}
      onSubmit={props.handleAddCoupon}
    >
      <label className={styles.label} htmlFor="coupon-code-field" style={props.labelStyle}>
        <I18n.Text string="cart.redeem_coupon" />
      </label>

      <Input
        className={styles.input}
        disabled={props.isLoading}
        name="coupon-code-field"
        onFocusChange={props.handleFocusChange}
        onChange={props.handleValueChange}
        value={props.value}
      />

      <div
        style={props.iconStyle}
        className={styles.icon}
        aria-hidden
        onClick={props.handleAddCoupon}
      >
        <AddIcon disabled={props.isButtonDisabled} />
      </div>
      <div className={styles.underlineWrapper}>
        <div className={`${styles.underline} ${(!props.isFocused || props.isLoading) ? styles.underlineBlurred : ''}`} />
      </div>
    </form>
  </div>
);

CouponFieldLayout.propTypes = {
  handleAddCoupon: PropTypes.func,
  handleFocusChange: PropTypes.func,
  handleValueChange: PropTypes.func,
  iconStyle: PropTypes.shape(),
  isButtonDisabled: PropTypes.bool,
  isFocused: PropTypes.bool,
  isLoading: PropTypes.bool,
  labelStyle: PropTypes.shape(),
  value: PropTypes.string,
};

CouponFieldLayout.defaultProps = {
  handleAddCoupon: () => {},
  handleFocusChange: () => {},
  handleValueChange: () => {},
  iconStyle: null,
  isButtonDisabled: false,
  isFocused: false,
  isLoading: false,
  labelStyle: null,
  value: '',
};

export default CouponFieldLayout;
