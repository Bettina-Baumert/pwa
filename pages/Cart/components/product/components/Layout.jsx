import React, { PropTypes } from 'react';
import { Grid } from 'Library/components';
import {
  ProductImage,
  QuantityPicker,
} from 'Templates/components';
import ProductTitle from './Title';
import ProductPrice from './Price';
import ProductProperties from './Properties';
import styles from '../style';

/**
 * The ProductLayout component.
 * @param {Object} props The component properties.
 * @returns {JSX}
 */
const ProductLayout = props => (
  <Grid className={styles.item}>
    <Grid.Item className={styles.leftColumn}>
      <div className={styles.image}>
        <ProductImage src={props.product.featuredImageUrl} />
      </div>
      <QuantityPicker
        quantity={props.quantity}
        editMode={props.editMode}
        onChange={props.handleUpdate}
        onToggleEditMode={props.toggleEditMode}
      />
    </Grid.Item>
    <Grid.Item className={styles.content} grow={1}>
      <ProductTitle
        handleRemove={props.handleDelete}
        toggleEditMode={props.toggleEditMode}
        value={props.product.name}
      />
      <Grid className={styles.info}>
        <ProductProperties
          properties={props.product.properties}
        />
        <ProductPrice
          currency={props.currency}
          defaultPrice={props.product.price.default}
          specialPrice={props.product.price.special}
        />
      </Grid>
    </Grid.Item>
  </Grid>
);

ProductLayout.propTypes = {
  currency: PropTypes.string.isRequired,
  editMode: PropTypes.bool.isRequired,
  product: PropTypes.shape().isRequired,
  quantity: PropTypes.number.isRequired,
  handleDelete: PropTypes.func,
  handleUpdate: PropTypes.func,
  toggleEditMode: PropTypes.func,
};

ProductLayout.defaultProps = {
  handleDelete: () => {},
  handleUpdate: () => {},
  toggleEditMode: () => {},
};

export default ProductLayout;
