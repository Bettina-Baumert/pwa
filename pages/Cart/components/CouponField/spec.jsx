import React from 'react';
import { mount } from 'enzyme';
import mockRenderOptions from '@shopgate/pwa-common/helpers/mocks/mockRenderOptions';
import CouponField from './index';
import NotSupported from './components/NotSupported';
import Layout from './components/Layout';

// Mock the redux connect() method instead of providing a fake store.
jest.mock('./connector', () => obj => obj);

describe('CouponField', () => {
  it('should render as expected without any props', () => {
    const wrapper = mount(<CouponField />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Layout).length).toBe(1);
    expect(wrapper.find(NotSupported).length).toBe(0);
  });

  it('should render a message when the cart supports coupons', () => {
    const wrapper = mount(<CouponField isSupported />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Layout).length).toBe(1);
    expect(wrapper.find(NotSupported).length).toBe(0);
  });

  it('should render a message when the cart does not support coupons', () => {
    const wrapper = mount(<CouponField isSupported={false} />, mockRenderOptions);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Layout).length).toBe(0);
    expect(wrapper.find(NotSupported).length).toBe(1);
  });
});
