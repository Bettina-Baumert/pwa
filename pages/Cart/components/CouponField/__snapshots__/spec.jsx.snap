// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CouponField should render a message when the cart does not support coupons 1`] = `
<MessageBar
  classNames={
    Object {
      "container": "",
      "message": "",
    }
  }
  messages={
    Array [
      Object {
        "message": "cart.coupons_not_supported",
        "type": "info",
      },
    ]
  }
/>
`;

exports[`CouponField should render a message when the cart supports coupons 1`] = `
<div>
  <Layout
    handleAddCoupon={[Function]}
    handleFocusChange={[Function]}
    handleValueChange={[Function]}
    iconStyle={
      Object {
        "opacity": 0,
      }
    }
    isButtonDisabled={true}
    isFocused={false}
    isLoading={false}
    labelStyle={
      Object {
        "display": "block",
      }
    }
    value=""
  />
</div>
`;

exports[`CouponField should render as expected without any props 1`] = `
<div>
  <Layout
    handleAddCoupon={[Function]}
    handleFocusChange={[Function]}
    handleValueChange={[Function]}
    iconStyle={
      Object {
        "opacity": 0,
      }
    }
    isButtonDisabled={true}
    isFocused={false}
    isLoading={false}
    labelStyle={
      Object {
        "display": "block",
      }
    }
    value=""
  />
</div>
`;
